SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -83,6 +83,126 @@ void UpdateChangelistHistory(TUniquePtr<FRepState>& RepState)
 }
 } // end anonymous namespace
 
+bool FSpatialObjectRepState::MoveMappedObjectToUnmapped_r(const FUnrealObjectRef& ObjRef, FObjectReferencesMap& ObjectReferencesMap)
+{
+	bool bFoundRef = false;
+
+	for (auto& ObjReferencePair : ObjectReferencesMap)
+	{
+		FObjectReferences& ObjReferences = ObjReferencePair.Value;
+
+		if (ObjReferences.Array != NULL)
+		{
+			if (MoveMappedObjectToUnmapped_r(ObjRef, *ObjReferences.Array))
+			{
+				bFoundRef = true;
+			}
+			continue;
+		}
+
+		if (ObjReferences.MappedRefs.Contains(ObjRef))
+		{
+			ObjReferences.MappedRefs.Remove(ObjRef);
+			ObjReferences.UnresolvedRefs.Add(ObjRef);
+			bFoundRef = true;
+		}
+	}
+
+	return bFoundRef;
+}
+
+bool FSpatialObjectRepState::MoveMappedObjectToUnmapped(const FUnrealObjectRef& ObjRef)
+{
+	if (MoveMappedObjectToUnmapped_r(ObjRef, ReferenceMap))
+	{
+		UnresolvedRefs.Add(ObjRef);
+		return true;
+	}
+	return false;
+}
+
+void FSpatialObjectRepState::GatherObjectRef(TSet<FUnrealObjectRef>& OutReferenced, TSet<FUnrealObjectRef>& OutUnresolved, const FObjectReferences& CurReferences) const
+{
+	if (CurReferences.Array)
+	{
+		for (auto const& Entry : *CurReferences.Array)
+		{
+			GatherObjectRef(OutReferenced, OutUnresolved, Entry.Value);
+		}
+	}
+
+	OutUnresolved.Append(CurReferences.UnresolvedRefs);
+	OutReferenced.Append(CurReferences.UnresolvedRefs);
+	OutReferenced.Append(CurReferences.MappedRefs);
+}
+
+void FSpatialObjectRepState::UpdateRefToRepStateMap(FObjectToRepStateMap& RepStateMap)
+{
+	// Inspired by FObjectReplicator::UpdateGuidToReplicatorMap
+	UnresolvedRefs.Empty();
+
+	TSet< FUnrealObjectRef > LocalReferencedObj;
+	for (auto& Entry : ReferenceMap)
+	{
+		GatherObjectRef(LocalReferencedObj, UnresolvedRefs, Entry.Value);
+	}
+
+	// TODO : Support references in structures updated by deltas.
+	//UObject* Object = GetObject();
```

Comment:
Appreciate you've done some investigation work, but we don't generally comment out blocks of code. Could you please create a tracking ticket, and move the code there. Thanks!